weave/integrations/integration_utilities.py

```diff
@@ -0,0 +1,67 @@
+import hashlib
+import re
+
+MAX_RUN_NAME_LENGTH = 128
+
+
+def make_pythonic_function_name(name: str) -> str:
+    name = name.replace("<", "_").replace(">", "")
+
+    valid_run_name = re.sub(r"[^a-zA-Z0-9 .\\-_]", "_", name)
+    return valid_run_name
+
+
+def truncate_op_name(name: str) -> str:
+    if len(name) <= MAX_RUN_NAME_LENGTH:
+        return name
+
+    trim_amount_needed = len(name) - MAX_RUN_NAME_LENGTH
+    parts = name.split(".")
+    last_part = parts[-1]
+
+    if len(last_part) <= trim_amount_needed:
+        # In this case, the last part is shorter than the amount we need to trim.
+        raise ValueError("Unable to create a valid run name from: " + name)
```

Comment:
Hmm, couldn't we do something else here?  I suppose we only hit this case if we had a MEGA long function name which should be very rare so probably it's ok.

Reply:
this should never happen in practice, since the first part is hard coded to `langchain.XYZ` where `XYZ` is a short classname.

Reply:
I see, I misread this initially.  This would blow up if a user had a really long parts of the string at the beginning. 👍